ginivision/src/doc/source/guide-for-open-with.rst

```diff
@@ -0,0 +1,182 @@
+ 
+The Intent for opening the *App info* page can be created as follows:
+
+.. code-block:: java
+
+    Intent intent = new Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
+    Uri uri = Uri.fromParts(Uri."package", BuildConfig.APPLICATION_ID, null);
+    intent.setData(uri);
+    startActivity(intent);
+
+Documentation
+^^^^^^^^^^^^^
+
+- https://developer.android.com/training/permissions/index.html
+
+Handling incoming PDFs and images
+---------------------------------
+
+When your app is requested to handle a PDF or an image your *Activity* (declaring the intent filter we mentioned previously) is launched or resumed with an Intent having ``ACTION_VIEW`` or ``ACTION_SEND``.
+
+Checking whether the Intent has the required action:
+
+.. code-block:: java
+
+    String action = intent.getAction();
+    if (Intent.ACTION_VIEW.equals(action) || Intent.ACTION_SEND.equals(action)) {
+        ...
+    }
+ 
+The Intent will contain an ``Uri`` pointing to the PDF or image. The ``Uri`` can be in the ``data`` field or since Android 4.1 (API Level 16) it can also be in the ``clipData`` field. 
+
+.. note:: 
+
+    To also support `ACTION_VIEW <https://developer.android.com/reference/android/content/Intent.html#ACTION_VIEW>`_ we recommend using ``data`` and ``clipData``. The `Androd example <https://developer.android.com/training/sharing/receive.html>`_ uses ``intent.getParcelableExtra(Intent.EXTRA_STREAM)`` to extract the ``Uri``, but this works only with `ACTION_SEND <https://developer.android.com/reference/android/content/Intent.html#ACTION_SEND>`_. 
+
+Getting the ``Uri`` from the Intent:
+
+.. code-block:: java
+
+    Uri uri = intent.getData();
+    if (uri == null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+        ClipData clipData = intent.getClipData();
+        if (clipData != null && clipData.getItemCount() > 0) {
+            uri = clipData.getItemAt(0).getUri();
+        }
+    }
+
+We recommend checking the MIME type of Intent. The MIME type can be in the ``type`` field or since Android 4.1 (API Level 16) it can also be in the ``clipData`` field.
+ 
+Getting the MIME type from the Intent:
+
+.. code-block:: java
+
+    List<String> mimeTypes = new ArrayList<>();
+    String type = intent.getType();
+    if (mimeType != null) {
+        mimeTypes.add(type);
+    } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+        ClipData clipData = intent.getClipData();
+        if (clipData != null) {
+            ClipDescription description = clipData.getDescription();
+            for (int i = 0; i < description.getMimeTypeCount(); i++) {
+                type = description.getMimeType(i);
+                mimeTypes.add(type);
+            }
+        }
+    }
+
+Having the ``Uri`` the contents of the PDF or image files can be read using the ``ContentResolver``.
```

Comment:
Suggest adding a comma after "Uri". So: "Having the 'Uri', the contents of the PDF or image files can be read using the 'ContentResolver'."